src/main/java/com/bobocode/creditadvisoryapp/model/Advisor.java

```diff
@@ -0,0 +1,65 @@
+package com.bobocode.creditadvisoryapp.model;
+
+import com.bobocode.creditadvisoryapp.enums.AppStatus;
+import com.bobocode.creditadvisoryapp.enums.Role;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.FetchType;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+@Getter
+@Setter
+@ToString(callSuper = true)
+@Entity
+@Table(name = "advisors")
+public class Advisor extends User {
+
+  @Enumerated(EnumType.STRING)
+  private Role role;
+
+  @OneToMany(mappedBy = "advisor", cascade = {CascadeType.PERSIST, CascadeType.REMOVE}, orphanRemoval = true)
+  @ToString.Exclude
+  @Setter(AccessLevel.PRIVATE)
+  private List<Application> apps = new ArrayList<>();
+
+  public void addApplication(Application app) {
+    apps.add(app);
+    app.setAdvisor(this);
+  }
+
+  public void removeApp(Application app) {
+    apps.remove(app);
+    app.setAdvisor(null);
+  }
+
+  public void validateAssignment() {
```

Comment:
This logic seems to be more related to the `Application` domain than to advisor. Since the `application` status is a responsibility of the `application`, I would move this logic to the `applications` side and do something like this in the `Application` class 
```
 public boolean isAssigned() {
          return status == assigned;
      }
```

 ```
  public void assignTo(Advisor advisor) {
          if (isAssigned()) {
              throw new RuntimeException();
          } else{ ..... }
```

and call `assignTo()` method here, passing `this` as an argument. 
What do you think?